[ENG-8954][docs] EAS Update: add pitch and revamp intro page (#23095)
expo/expo: docs/pages/eas-update/codepush.mdx

import { Terminal } from '~/ui/components/Snippet';
import { Step } from '~/ui/components/Step';

> This guide explains how to transition a React Native project that uses CodePush to use EAS Update, offering [many advantages](/eas-update/introduction/#pitch). It assumes that you are using the default React Native project structure. For assistance with migrating brownfield native apps to EAS Update, [reach out to us directly](https://expo.dev/contact).

Reviewer: ```suggestion
This guide explains how to transition a React Native project that uses CodePush to use EAS Update which offers [many advantages](/eas-update/introduction/#pitch). It assumes that you're using the default React Native project structure. For assistance with migrating brownfield native apps to EAS Update, [reach out to us directly](https://expo.dev/contact).
```

- Suggestion to rephrase offering phrase
